Realm/Realm.Sync/SyncConfiguration.cs

```diff
@@ -87,11 +109,18 @@ internal override Realm CreateRealm(RealmSchema schema)
 
         private Native.SyncConfiguration ToNative()
         {
+            if (!string.IsNullOrEmpty(TrustedCAPath) &&
+                !File.Exists(TrustedCAPath))
+            {
+                throw new RealmFileNotFoundException($"File not found: {TrustedCAPath}");
```

Comment:
I agree - we throw a bunch of system exceptions anyway. I'll update it here, but will leave the rest of the codebase for another time ;)